aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/ReachabilityMonitor.kt

```diff
@@ -91,39 +87,51 @@ private class ReachabilityMonitorImpl constructor(val schedulerProvider: Schedul
         }
         return subject.subscribeOn(schedulerProvider.io())
     }
+
+    private inner class ConnectivityNetworkCallback(private val emitter: ObservableEmitter<Boolean>) : NetworkCallback() {
+        override fun onAvailable(network: Network) {
+            connectivityProvider?.let { emitter.onNext(it.hasActiveNetwork) }
+        }
+
```

Comment:
For onAvailable method, should we just reuse the previous networkCapabilities if it exists so it could be avoid calling hasActiveNetwork method which is more expensive. 

```
private inner class ConnectivityNetworkCallback(private val emitter: ObservableEmitter<Boolean>) : NetworkCallback() {
       private var currentNetworkCapabilities: NetworkCapabilities? = null
        override fun onAvailable(network: Network) {
            currentNetworkCapabilities?.let { networkCapabilities ->
                emitter.onNext(networkCapabilities.isInternetReachable())
            } ?:  connectivityProvider?.let { 
                emitter.onNext(it.hasActiveNetwork) 
            }
           
        }

        override fun onLost(network: Network) {
            emitter.onNext(false)
        }

        override fun onUnavailable() {
            emitter.onNext(false)
        }

        override fun onCapabilitiesChanged(network: Network, networkCapabilities: NetworkCapabilities) {
            currentNetworkCapabilities = networkCapabilities
            emitter.onNext(networkCapabilities.isInternetReachable())
        }
    }
```

Reply:
Updated this change in the code. Please let me know if it need any modification.